src/components/UNSTABLE_Pagination/PageSelector.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file Page selector.
+ * @copyright IBM Security 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Select, SelectItem } from '../Select';
+import { appendComponentNamespace } from '../../globals/namespace/index';
+import { namespace as paginationNamespace } from './Pagination';
+
+const namespace = appendComponentNamespace(
+  paginationNamespace,
+  'page-selector'
+);
+
+function PageSelector({ className, id, labelText, totalPages, ...rest }) {
```

Comment:
I believe we can reduce and refactor the logic here by doing something like the following:

```js
return [...Array(total).keys()].map(counter => <SelectItem key={counter} text={String(counter)} value={counter} />)`
```

In this case, we won't need to toggle off the `no-plusplus` rule either.

Reply:
Thank you! I'll refactor these lines... but I'm hesitant about the suggestion above for a couple reasons...
- main thing to me is that it is harder to read this way... spreading the `Array.keys()` iterator object like that seems a little obscure to me? And seeing a `map` there, at first, was a little confusing because, in context of what's going on here, we are trying to use a counter to create an array, not start with an anonymous array and create a different array.
- counter starts at 0, so we'll need to put counter + 1 in every instance of counter. Doesn't really change a whole lot but it would make it all more dense/repetitive, if we wanted to keep it a single line
- smaller point: `keys={counter}` may go against the React eslint requirement that you don't use an index in a key? 🤔

Reply:
I'm not strongly opposed to this approach, as they're both essentially doing the same thing. My only concern with this is that we're switching off a linting rule to achieve something that can be achieved with an alternative solution.

I personally find traditional `for` / `while` loops to be more error-prone and would advocate avoiding these as much as possible in favor of using built-in array methods like `map`.

Reply:
@SimonFinney I think this is relevant to this style PR I noticed this morning, about writing code explicitly: `https://github.com/carbon-design-system/carbon/pull/4879`

I'm not familiar with error prone nature of `for` / `while` loops? I was thinking the opposite... that using `Array.keys()` in this manner could be more error prone (for future devs looking at the code in particular).

Reply:
That's a very good point - this is a much more valuable approach for future debugging and something I hadn't entirely considered. I'm torn between us being able to optimize productivity while also leveraging newer features of the language. I'm on board!